pkg/clean/clean_dlb.go

```diff
@@ -0,0 +1,147 @@
+package clean
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/config"
+)
+
+type (
+	mapInterface = map[string]interface{}
+	mapBool      = map[string]bool
+)
+
+func httpGet(cf *config.NSXOperatorConfig, url string) (map[string]interface{}, error) {
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+
+	log.Info("Getting ", "url", url)
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return nil, err
+	}
+	req.SetBasicAuth(cf.NsxApiUser, cf.NsxApiPassword)
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer func(Body io.ReadCloser) {
+		err := Body.Close()
+		if err != nil {
+			fmt.Println("Failed to close response body")
+		}
+	}(resp.Body)
+
+	var response mapInterface
+	err = json.NewDecoder(resp.Body).Decode(&response)
+	if err != nil {
+		return nil, err
+	}
+	return response, nil
+}
+
+func httpDelete(cf *config.NSXOperatorConfig, url string) error {
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+
+	log.Info("Deleting ", "url", url)
+	req, err := http.NewRequest("DELETE", url, nil)
+	if err != nil {
+		return err
+	}
+	req.SetBasicAuth(cf.NsxApiUser, cf.NsxApiPassword)
+
+	_, err = client.Do(req)
+	return err
+}
+
+func checkTagsExist(tags []interface{}) bool {
+	requiredTags := mapBool{"ncp/version": false, "ncp/cluster": false}
+
+	for _, tagItem := range tags {
+		if scope, ok := tagItem.(mapInterface)["scope"].(string); ok {
+			if _, exists := requiredTags[scope]; exists {
+				requiredTags[scope] = true
+			}
+		}
+	}
+
+	for _, exists := range requiredTags {
+		if !exists {
+			return false
+		}
+	}
+	return true
+}
+
+func httpGetDLBServices(cf *config.NSXOperatorConfig) ([]string, error) {
+	url := "https://" + cf.NsxApiManagers[0] + ":443/policy/api/v1/infra/lb-services/"
+	resp, err := httpGet(cf, url)
+	if err != nil {
+		return nil, err
+	}
+
+	var dlbServicesPath []string
+	for _, item := range resp["results"].([]interface{}) {
+		if item.(mapInterface)["size"].(string) == "DLB" && checkTagsExist(item.(mapInterface)["tags"].([]interface{})) {
+			dlbServicesPath = append(dlbServicesPath, item.(mapInterface)["path"].(string))
+		}
+	}
+	return dlbServicesPath, nil
+}
+
+func httpGetVirtualServers(cf *config.NSXOperatorConfig, dlbServicesPath []string) ([]string, []string, error) {
+	url := "https://" + cf.NsxApiManagers[0] + ":443/policy/api/v1/infra/lb-virtual-servers/"
```

Comment:
can use cluster tag as query when getting lb-virtual-servers to only retrieve virtual servers for DLB.

Reply:
ok